Analytics/Analytics/StateManagement/UserIdentity.swift

```diff
@@ -10,40 +10,141 @@ import Foundation
 // MARK: - UserIdentity
 /**
  A struct representing the identity of a user.
+ 
+ This structure is used to manage the identification of a user, including both anonymous and logged-in states, as well as associated traits and external identifiers.
 
  - Properties:
    - `anonymousId`: A unique identifier for the user when they are not logged in. Defaults to an empty string.
+   - `userId`: The identifier for the user when they are logged in. Defaults to an empty string.
+   - `traits`: A dictionary of user-specific traits, used to store additional metadata about the user.
+   - `externalIds`: An array of external identifiers associated with the user, allowing integration with external systems or platforms.
 
  - Methods:
    - `initializeState(_:)`: Creates and initializes a `UserIdentity` instance by reading stored data from a key-value storage.
  */
-
-struct UserIdentity {
-    /// A unique identifier for the user when they are not logged in.
-    var anonymousId = ""
-
+public struct UserIdentity {
+    /// A unique identifier for the user when they are not logged in. Defaults to an empty string.
+    public internal(set) var anonymousId = ""
+    
+    /// The identifier for the user when they are logged in. Defaults to an empty string.
+    public internal(set) var userId = ""
+    
+    /// A dictionary of user-specific traits, used to store additional metadata about the user.
+    public internal(set) var traits = RudderTraits()
+    
+    /// An array of external identifiers associated with the user, allowing integration with external systems or platforms.
+    public internal(set) var externalIds = [ExternalId]()
+    
     /**
      Creates and initializes a `UserIdentity` instance by reading data from the provided key-value storage.
-
+     
      - Parameter storage: An instance of `KeyValueStorage` used to read the stored user identity data.
      - Returns: A fully initialized `UserIdentity` object with data from the storage or default values if not available.
+     
+     The method performs the following steps:
+     1. Reads the `anonymousId` from storage or generates a new UUID.
+     2. Reads the `userId` from storage or sets it to an empty string.
+     3. Parses and assigns user traits from the stored JSON string, if available.
+     4. Decodes and assigns external IDs from the stored array of JSON strings.
      */
+    
     static func initializeState(_ storage: KeyValueStorage) -> UserIdentity {
         var identity = UserIdentity()
-
+        
         identity.anonymousId = storage.read(key: StorageKeys.anonymousId) ?? .randomUUIDString
-
+        identity.userId = storage.read(key: StorageKeys.userId) ?? ""
```

Comment:
Define an extension property called `String.empty = ""` and use it everywhere instead of hard coding `""`. In this way, we can find all the places where we use empty string and it'll also be in parity with Kotlin.

```suggestion
        identity.userId = storage.read(key: StorageKeys.userId) ?? String.empty
```

Reply:
Done